application-workloads/azure-gamedev/gamedev-vm/scripts/parsec/Clear-Proxy.ps1

```diff
@@ -0,0 +1,271 @@
+Elseif($gpu.Device_ID -eq "DEV_1EB8") {$gpu.Name = 'NVIDIA Tesla T4'; $gpu.PSID = '110'; $gpu.PFID = '883'; $gpu.NV_GRID = $true; $gpu.Driver_Version = driverversion; $gpu.Web_Driver = webdriver; $gpu.Update_Available = ($gpu.Web_Driver -gt $gpu.Driver_Version); $gpu.Current_Mode = GPUCurrentMode; $gpu.Supported = "UnOfficial"}
+Elseif($gpu.Device_ID -eq $null) {$gpu.Supported = "No"; $gpu.Name = "No Device Found"}
+else{$gpu.Supported = "No"; $gpu.Name = webName}
+}
+
+function checkGPUSupport{
+#quits if GPU isn't supported
+If ($gpu.Supported -eq "No") {
+$app.FailGPU
+Exit
+}
+ElseIf ($gpu.Supported -eq "UnOfficial") {
+$app.UnOfficialGPU
+}
+Else{}
+}
+
+function checkDriverInstalled {
+#Tells user if no GPU driver is installed
+if ($system.Valid_NVIDIA_Driver -eq $False) {
+$app.NoDriver
+}
+Else{}
+}
+
+function prepareEnvironment {
+#prepares working directory
+$test = Test-Path -Path $system.path 
+if ($test -eq $true) {
+Remove-Item -path $system.Path -Recurse -Force | Out-Null
+New-Item -ItemType Directory -Force -Path $system.path | Out-Null}
+Else {
+New-Item -ItemType Directory -Force -Path $system.path | Out-Null
+}
+}
+
+function startUpdate { 
+#Gives user an option to start the update, and sends messages to the user
+       prepareEnvironment
+       downloaddriver
+       InstallDriver
+       rebootlogic
+}
+
+function setnvsmi {
+#downloads script to set GPU to WDDM if required
+(New-Object System.Net.WebClient).DownloadFile("https://raw.githubusercontent.com/parsec-cloud/Cloud-GPU-Updater/master/Additional%20Files/NVSMI.ps1", $($system.Path) + "\NVSMI.ps1") 
+Unblock-File -Path "$($system.Path)\NVSMI.ps1"
+}
+
+function setnvsmi-shortcut{
+#creates startup shortcut that will start the script downloaded in setnvsmi
+Write-Output "Create NVSMI shortcut"
+$Shell = New-Object -ComObject ("WScript.Shell")
+$ShortCut = $Shell.CreateShortcut("$env:PROGRAMDATA\Microsoft\Windows\Start Menu\Programs\Startup\NVSMI.lnk")
+$ShortCut.TargetPath="powershell.exe"
+$ShortCut.Arguments='-WindowStyle hidden -ExecutionPolicy Bypass -File "C:\ParsecTemp\Drivers\NVSMI.ps1"'
+$ShortCut.WorkingDirectory = "C:\ParsecTemp\Drivers";
+$ShortCut.WindowStyle = 0;
+$ShortCut.Description = "Create NVSMI shortcut";
+$ShortCut.Save()
+}
+
+function DownloadDriver {
+if (($gpu.supported -eq "UnOfficial") -eq $true) {
+(New-Object System.Net.WebClient).DownloadFile($($($URL.GoogleGRID).links | Where-Object href -like *server2016_64bit_international.exe*).href, "C:\ParsecTemp\Drivers\GoogleGRID.exe")
+}
+Else {
+#downloads driver from nvidia.com
+$Download.Link = Invoke-WebRequest -Uri $gpu.url -Method Get -UseBasicParsing | select @{N='Latest';E={$($_.links.href -match"www.nvidia.com/download/driverResults.aspx*")[0].substring(2)}}
```

Comment:
similar to the 7zip pattern, feels fragile, though if there is more of a need to get the latest drivers is probably ok...

Reply:
hardened with suggestions from https://github.com/lord-carlos/nvidia-update/blob/master/nvidia.ps1